src/videojs-media-sources.js

```diff
@@ -100,64 +98,166 @@
+            (this.audioBuffer_ && this.audioBuffer_.updating) ||
+            (this.videoBuffer_ && this.videoBuffer_.updating);
         }
       });
       // the buffered property is the intersection of the buffered
       // ranges of the native source buffers
       Object.defineProperty(this, 'buffered', {
         get: function() {
-          var start, end;
-          if (this.videoBuffer_.buffered.length === 0 ||
-              this.audioBuffer_.buffered.length === 0) {
+          var
+            start,
+            end,
+            starts = [],
+            ends = [],
+            lastIndex;
+
+          // Handle the case where there is no buffer
+          if ((this.videoBuffer_ && this.videoBuffer_.buffered.length === 0) ||
+              (this.audioBuffer_ && this.audioBuffer_.buffered.length === 0)) {
             return videojs.createTimeRange();
           }
-          start = Math.max(this.videoBuffer_.buffered.start(0),
-                           this.audioBuffer_.buffered.start(0));
-          end = Math.min(this.videoBuffer_.buffered.end(0),
-                         this.audioBuffer_.buffered.end(0));
+
+          // TODO: Calculate and return the true intersection of the buffered
+          //       property
+          if (this.videoBuffer_) {
+            lastIndex = this.videoBuffer_.buffered.length - 1;
+            starts.push(this.videoBuffer_.buffered.start(0));
+            ends.push(this.videoBuffer_.buffered.end(lastIndex));
+          }
+          if (this.audioBuffer_) {
+            lastIndex = this.audioBuffer_.buffered.length - 1;
+            starts.push(this.audioBuffer_.buffered.start(0));
+            ends.push(this.audioBuffer_.buffered.end(lastIndex));
+          }
+
+          start = Math.max.apply(Math, starts);
+          end = Math.min.apply(Math, ends);
           return videojs.createTimeRange(start, end);
         }
       });
     },
     appendBuffer: function(segment) {
-      this.audioUpdating_ = this.videoUpdating_ = true;
-      this.transmuxer_.push(segment);
-      this.transmuxer_.flush();
+      // Start the internal "updating" state
+      this.bufferUpdating_ = true;
+
+      this.transmuxer_.postMessage({action: 'push', data: segment.buffer}, [segment.buffer]);
+      this.transmuxer_.postMessage({action: 'flush'});
+    },
+    /**
+     * Process any segments that the muxer has output
+     * Concatenate segments together based on type and append them into
+     * their respective sourceBuffers
+     */
+    processPendingSegments_: function() {
+      var sortedSegments = {
+          video: {
+            segments: [],
+            bytes: 0
+          },
+          audio: {
+            segments: [],
+            bytes: 0
+          }
+        };
+
+      // Sort segments into separate video/audio arrays and
+      // keep track of their total byte lengths
+      sortedSegments = this.pendingBuffers_.reduce(function (segmentObj, segment) {
```

Comment:
Reduce doesn't seem to be available in IE8, maybe you can add a babel step if you want the code to be cleaner